nodego/events.go

```diff
@@ -0,0 +1,161 @@
+// Copyright 2018 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package nodego
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"net/http"
+	"runtime/debug"
+	"time"
+
+	"google.golang.org/api/pubsub/v1"
```

Comment:
I just remembered that I requested before that the code that depends on these be made a separate, optional package. It seems like this file (and maybe the new trigger constants) could be moved to a new events package or something without causing much trouble. Regardless, please add them as dependencies in the REAME and install them in the [provision](https://github.com/GoogleCloudPlatform/cloud-functions-go/blob/master/provision.sh) script.

Reply:
I'm not sure what this means. If I move it to a sub package, then would it not already be accessible in the virtual environment?

Reply:
Those are separate requests:
1. Move it to an optional subpackage. This means that users who only use HTTP triggers don't need these dependencies. This is mostly useful for existing users who update the library.
2. Add it to the list of dependencies so that new users install it and thus can use the new triggers.
3. Add it to the provision script so that new environments created with Vagrant have the new dependencies and thus work with the new triggers.

Reply:
I'm getting confused at "list of dependencies". Does this mean non-stdlib dependencies like `google.golang.org/api/pubsub/v1` and `google.golang.org/api/storage/v1`?

If that's the case, and I'm understanding correctly, you'd like a new section in the readme that explicitly states that these are imported in the events package as well as a line in the provision script to `go get` them?

Reply:
README: Just add it in [Requirements](https://github.com/GoogleCloudPlatform/cloud-functions-go#requirements).

provision: yes, exactly